db/db.go

```diff
@@ -211,18 +218,26 @@ func (db *Connection) configure(maxIdleConns int, maxOpenConns int) {
 	db.gennericDB.SetMaxOpenConns(maxOpenConns)
 }
 
-func (db *Connection) setupMetrics() {
-	// ping to check status
-	pingStop := doEvery(time.Second, func() {
-		err := db.Ping()
-		if err != nil {
-			db.measures.ConnectionStatus.Set(0.0)
-		} else {
-			db.measures.ConnectionStatus.Set(1.0)
-		}
+func (db *Connection) setupHealthCheck() {
+	if db.health == nil {
+		return
+	}
+	sqlCheck, err := checkers.NewSQL(&checkers.SQLConfig{
+		Pinger: db.gennericDB,
 	})
-	db.stopThreads = append(db.stopThreads, pingStop)
+	if err != nil {
+		// todo: capture this error somehow
+	}
 
+	db.health.AddCheck(&health.Config{
+		Name:     "sql-check",
+		Checker:  sqlCheck,
+		Interval: time.Duration(1) * time.Second,
```

Comment:
I believe we should use the ping interval from the config. 	https://github.com/Comcast/codex/blob/master/db/db.go#L56